src/MicrosoftTeams.ts

```diff
@@ -47,6 +55,191 @@ namespace microsoftTeams {
+     */
+    export interface ViewData {
+      /**
+       * Display header title of the item list.
+       */
+      listTitle?: string;
+
+      /**
+       * Type of the menu item.
+       */
+      listType: MenuListType;
+
+      /**
+       * Array of MenuItem. Icon value will be required for all items in the list.
+       */
+      listItems: MenuItem[];
+    }
+
+    /**
+     * Represents information about type of list to display in Navigation Bar Menu.
+     */
+    export enum MenuListType {
+      dropDown = "dropDown",
+      popOver = "popOver"
+    }
+
+    let navBarMenuItemPressHandler: (id: String) => boolean;
+    handlers["navBarMenuItemPress"] = handleNavBarMenuItemPress;
+
+    let actionMenuItemPressHandler: (id: String) => boolean;
+    handlers["actionMenuItemPress"] = handleActionMenuItemPress;
+
+    let viewConfigItemPressHandler: (id: String) => boolean;
+    handlers["setModuleView"] = handleViewConfigItemPress;
+
+    /**
+     * Registers list of view configurations and it's handler.
+     * Handler is responsible for listening selection of View Configuration.
+     * @param viewConfig List of view configurations. Minimum 1 value is required.
+     * @param handler The handler to invoke when the user selects view configuration.
+     */
+    export function setUpViews(
+      viewConfig: ViewConfiguration[],
+      handler: (id: string) => boolean
+    ): void {
+      ensureInitialized();
+      viewConfigItemPressHandler = handler;
+      sendMessageRequest(parentWindow, "setUpViews", [viewConfig]);
+    }
+
+    function handleViewConfigItemPress(id: String): void {
+      if (!viewConfigItemPressHandler || !viewConfigItemPressHandler(id)) {
+        ensureInitialized();
+        sendMessageRequest(parentWindow, "viewConfigItemPress", [id]);
+      }
+    }
+
+    /**
+     * Used to set menu items on the Navigation Bar. If icon is available, icon will be shown, otherwise title will be shown.
+     * @param items List of MenuItems for Navigation Bar Menu.
+     * @param handler The handler to invoke when the user selects menu item.
+     */
+    export function setNavBarMenu(
+      items: MenuItem[],
+      handler: (id: string) => boolean
+    ): void {
+      ensureInitialized();
+
+      navBarMenuItemPressHandler = handler;
+      sendMessageRequest(parentWindow, "setNavBarMenu", [items]);
+    }
+
+    function handleNavBarMenuItemPress(id: String): void {
+      if (!navBarMenuItemPressHandler || !navBarMenuItemPressHandler(id)) {
+        ensureInitialized();
+        sendMessageRequest(parentWindow, "handleNavBarMenuItemPress", [id]);
+      }
+    }
+
+    /**
+     * Used to show Action Menu.
+     * @param title Title for Action Menu
+     * @param items List of MenuItems for Action Menu.
+     * @param handler The handler to invoke when the user selects menu item.
+     */
+    export function showActionMenu(
+      title: string,
+      items: MenuItem[],
```

Comment:
Might be better to wrap the title and items into their own class (e.g. ActionMenuConfiguration or ActionMenuParameters) so that we can easily add to them later without re-ordering the parameters of the API. For example, when we implement this in web/desktop we would need to ask the developer to provide relative coordinates for the menu.